### ISSUE TITLE
[HOLD for payment 2024-01-25] [$500] [Wave 6: Workspace Chats] IOU - Chat with IOU shows "This is beginning" in LHN after login and no Pay elsewhere in IOU preview

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#32658]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.9-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
Precondition: User has received IOU request
1. Log out
2. Log in
3. Open the chat with IOU request

## Expected Result:
The chat preview in LHN will show the latest IOU action
After opening the chat, the IOU preview shows Pay elsewhere

## Actual Result:
The chat preview in LHN shows "This is the beginning of" instead of the latest IOU action
After opening the chat, the IOU preview does not show Pay elsewhere unless the IOU report is visited and user returns to the main chat

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/57075e2d-fff9-4348-9f98-23ff512800b7

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ff0e7b70f9f8b4d4</li>
        <li>Upwork Job ID: 1732753199542329344</li>
        <li>Last Price Increase: 2023-12-07</li>
<li>Automatic offers: </li>
<ul>
<li>cubuspl42 | Reviewer | 28080970</li>
<li>paultsimura | Contributor | 28080971</li>
</ul></li>
    </ul>
</details>

**Comment by paultsimura** [comment]
This seems like a BE issue: on the `OpenApp` request, only chat reports are loaded. As a result, the ReportPreview actions show invalid data, since the IOU Reports are absent from Onyx.
This might be a side-effect of https://github.com/Expensify/App/issues/29593

**Comment by cubuspl42** [comment]
@thienlnam Would you give it a look?

**Comment by rezkiy37** [comment]
> CC: @puneetlath @rezkiy37 @situchan as well for the hunch it might be related to #29593

AFAIK, we didn't touch policy expense report chats.

**Comment by trjExpensify** [comment]
This looks like it's a preview of an iouReport inside a DM. Not a preview of an expenseReport inside a policyExpenseChat (A.K.A "workspace chat").

**Comment by puneetlath** [comment]
Adding this to the wave6 project as critical so that someone can investigate. cc @greg-schroeder

**Comment by yuwenmemon** [comment]
Seems like `hasOutstandingIOU` is being sent as `false` here...
<img width="1176" alt="Screenshot 2023-12-13 at 2 58 56 PM" src="https://github.com/Expensify/App/assets/4741899/7eac42e1-a5eb-4e87-9d15-b54fd11e2b57">

Curious if this may be fixed by some of the changes in https://github.com/Expensify/Auth/pull/9359 (The other bug I was working on was too 🤔) Going to test one sec...

**Comment by yuwenmemon** [comment]
It _does_ get us closer:
<img width="391" alt="Screenshot 2023-12-13 at 3 22 23 PM" src="https://github.com/Expensify/App/assets/4741899/547528f1-bf50-46d5-aa38-23ac0d75af68">

**Comment by yuwenmemon** [comment]
Okay, I actually think I have a fix that's purely front-end. [This](https://github.com/Expensify/App/blob/138e6a22a1f17ff04deb650098d54e385460fe04/src/libs/OptionsListUtils.js#L394) unnecessarily uses the reportID stored on a sometimes absent iou report. We have all the information to show what we need already because `getIOUReportIDFromReportActionPreview` returns us a the iouReportID which we can use instead.

**Comment by cubuspl42** [comment]
@yuwenmemon I've seen you mentioned `hasOutstandingIOU`, so I'll link this PR:

- #32653

**Comment by yuwenmemon** [comment]
Ah sorry @cubuspl42 I forgot to follow up. That ended up not being a factor in what was going wrong.

**Comment by yuwenmemon** [comment]
Okay circled back internally and following up from [this thread](https://expensify.slack.com/archives/C02MW39LT9N/p1702579124232819), this is my understanding of what is going wrong:

We grab the parent report of an IOU that is awaiting action from a user in [getChats](https://github.com/Expensify/Auth/blob/2f66bbf996e2bec33582d6b74b1a56ba4b08fad1/auth/lib/Report.cpp#L5099-L5129). However, we don't grab the IOU report itself and thus show an erroneous message on the LHN because the IOU report is not in Onyx. Additionally, because we don't load the IOU reports, we don't show the "Pay with Expensify Button" and even have missing data about the transactions show up on the report:
<img width="554" alt="Screenshot 2023-12-14 at 12 55 02 PM" src="https://github.com/Expensify/App/assets/4741899/b9591ab5-eef1-41da-a10f-5f90ae626491">

@mountiny @cead22 @luacmartins @puneetlath Curious how you think we should move forward... If we don't want to load IOU reports right away in `OpenApp`, we can rewrite [getLastMessageTestForReport](https://github.com/Expensify/App/blob/a24f77b27fce2d79302e2076a32461e595684183/src/libs/OptionsListUtils.js#L378-L424) to use the report action that links to the IOU report itself for the preview. And then load all IOU reports related to a Chat Report only when we _open_ the chat report?

**Comment by luacmartins** [comment]
That makes sense to me. We could probably update this logic to just get `message[0].html` from the reportPreview action.

**Comment by mountiny** [comment]
I am wondering though, we still need to load the iou report in Web in order to compose the report preview message components and add the state/ status report preview action etc, no?

**Comment by yuwenmemon** [comment]
Right, I believe all of those only need to be shown once you open the report, so should we only load those via `OpenReport`?

**Comment by yuwenmemon** [comment]
Also, just curious do we not want to use the `lastMessageText`/`Html`?
<img width="369" alt="Screenshot 2023-12-15 at 10 43 37 AM" src="https://github.com/Expensify/App/assets/4741899/ff92c290-b35e-4e73-aacc-212d6077be6a">

**Comment by mountiny** [comment]
> Right, I believe all of those only need to be shown once you open the report, so should we only load those via OpenReport?

Hmm what I meant is that when you open the workspace chat for example, we need the expense reports state and status  or total of all the linked expense reports that you can see the report preview of. For that we need to load the expense report from cache most likely. This also composes sense message of the report preview action so lastMessageText of the workspace chat if the report preview action is last one.

**Comment by yuwenmemon** [comment]
@mountiny I'm sorry I'm not sure I follow - so what are you proposing?

**Comment by yuwenmemon** [comment]
Okay, after discussing it internally with the team, this is not a back-end issue and this is instead a result of a purposeful reduction in data being sent from the back-end in `OpenApp`.

We have all the information we need to assemble the last message text without having loaded the IOU Report associated with the last message. We can use the `lastMessageText`/`lastMessageHtml` sent with the parent chat report:
<img width="1120" alt="Screenshot 2023-12-20 at 3 07 46 PM" src="https://github.com/Expensify/App/assets/4741899/bef88f4d-7e97-44ce-9dce-c8cd7bf5ad10">

Some considerations will need to be made to insert personal details and allow for translations (in other words, get "yuwen@expensify.com owes $0.65" to "Yuwen debe 0,65US$") - but the information we have is there without us needing to load the IOU reports right away.

So, I'm going to flip this back to External!

**Comment by yuwenmemon** [comment]
Then @paultsimura regarding your concern about the missing IOU details in the chat, we should again be able to re-assemble the proper report review using the details we have stored in the report action itself. But let's handle that in a separate issue and keep the scope here to just the LHN message text.
<img width="554" alt="290683843-b9591ab5-eef1-41da-a10f-5f90ae626491" src="https://github.com/Expensify/App/assets/4741899/5fd50435-cca0-4ded-ac86-2453424ec456">

**Comment by cubuspl42** [comment]
@yuwenmemon Thank you very much!

@paultsimura Would you like to update your proposal based on this information, utilizing `lastMessageText`/`lastMessageHtml` properties?

**Comment by paultsimura** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
After login, the last messages of reports are displayed incorrectly.

### What is the root cause of that problem?
It was decided to move away from loading all the reports data from BE, so this issue looks like a new feature request now.

### What changes do you think we should make in order to solve the problem?
First of all, the reports indeed hold fields like: `lastMessageHtml`, `lastMessageText`. However, these fields are not localizable and do not hold the necessary data to localize them appropriately.

IMO, it's worth adding a nested object (similar to `reportAction.originalMessage`) to the Report entity. Let's say, `Report.lastReportActionDetails`. And populate this object with all the needed information about the last report action: type, amount, actor, etc. Based on these details, we should be able to build a localized preview message similar to what we do when copying the `reportAction` to the clipboard, since the preview message is always tied to the last visible report actions.

And we could use the `report.lastMessageText` as a fallback value.

This will require quite a lot of changes in the optimistic flows as well as on the BE side. But I can't think of a better way to implement a proper localization. Using REGEX's to parse the messages and guess the translation key seems a bit wild for me, especially considering that we have all the necessary data to build these preview messages, and we can just return them in the response without loading extra reports.

### What alternative solutions did you explore? (Optional)
I've seen in some discussions, there were plans to implement BE-side localization for some types of messages. If it is still a thing, we can rely solely on the `report.lastMessageText` after this localization is implemented.

**Comment by cubuspl42** [comment]
I once heard that backend localization is "not happening soon"...

@yuwenmemon What are your thoughts about the localization concerns?

**Comment by yuwenmemon** [comment]
Yes, indeed it won't be happening soon. I believe we do something similar to a workaround for the `[Attachment]` message:
https://github.com/Expensify/App/blob/df981144e66748e796e3b6c7d72fa8c6e86a38cc/src/libs/OptionsListUtils.js#L408-L409

**Comment by paultsimura** [comment]
## Proposal
Updated proposal https://github.com/Expensify/App/issues/32658#issuecomment-1866134497

However, this looks more like a feature request to me 🤔

**Comment by rojiphil** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

After logout+login, LHN shows the message `This is the beginning…` for a Chat Report which has a Money Request Report Preview as the last visible action.

### What is the root cause of that problem?

Purposeful reduction in data being sent from BE in OpenApp is the root cause

### What changes do you think we should make in order to solve the problem?

To solve the problem, we can ensure that the displayed reports in LHN are fetched at the app launch along with all linked reports up to a single level depth. We can consider more than one level of depth if required. 
More specifically, we can leverage `optionListItems` [here](https://github.com/Expensify/App/blob/bcee124ce4efd5eee6171fe0a3f7f0fa5db8de85/src/pages/home/sidebar/SidebarLinksData.js#L75) and ensure that all these reports are fetched once at app launch. Additionally, we can look for `childReportID` in Report Actions of these reports which is a good indicator of linked reports, and fetch them. 

We can implement a side effect [here](https://github.com/Expensify/App/blob/bcee124ce4efd5eee6171fe0a3f7f0fa5db8de85/src/pages/home/sidebar/SidebarLinksData.js#L75) which looks something like this to solve the problem:

    const fetchedReportIDsRef = useRef([]);
    useEffect(() => {
        const arrayChildReportIDs = [];
        optionListItems.forEach((reportID) => {
            const reportActions = allReportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`];
            const childReportIDs = _.filter(_.pluck(reportActions, 'childReportID'),(childReportID) => childReportID);
            if(childReportIDs.length > 0) {
                arrayChildReportIDs.push(childReportIDs);
            } 
        });
        const allRecentReportIDs = _.union(optionListItems, ...arrayChildReportIDs);
        if(currentReportID) {
            allRecentReportIDs.push(currentReportID);
        }

        const unfetchedReportIDs = _.difference(allRecentReportIDs, fetchedReportIDsRef.current);
        unfetchedReportIDs.forEach((reportID) => {
            if(!reportID) {
                return;
            }
            Report.reconnect(reportID);
            fetchedReportIDsRef.current.push(reportID);
        });        
    }, [currentReportID, optionListItems, chatReports, betas, policies, priorityMode, allReportActions]);  

Additionally, we also need to include `childReportID` in `reportActionsSelector` [here](https://github.com/Expensify/App/blob/bcee124ce4efd5eee6171fe0a3f7f0fa5db8de85/src/pages/home/sidebar/SidebarLinksData.js#L180)

### What alternative solutions did you explore? (Optional)

**Comment by rojiphil** [comment]
@yuwenmemon @mountiny  @luacmartins  @cubuspl42 

> this is instead a result of a purposeful reduction in data being sent from the back-end in OpenApp.

Ah, here it is. The above comment seems to nail the root cause of the concern of bad offline experience for users which I had raised in ([slack thread here](https://expensify.slack.com/archives/C01GTK53T8Q/p1702369258137099)). As I understand, the LHN description issue here is just one consequence of the reduced data being sent from BE. I think there are many more such consequences in FE to get revealed with time. To understand the impact of this change in BE, we can just go offline after signing in as a user. And, we will see the impact as shown in the test video that was shared in slack. 

Now, from the solution perspective, I think it is better to address this at the core and the issue here will also get addressed. But, if the decision is to solve this somewhere else as mentioned [here](https://github.com/Expensify/App/issues/32658#issuecomment-1865271421), we can consider my proposal there. 

To solve this, I think we can ensure that the displayed reports in LHN are fetched at app launch along with all linked reports upto a single level depth. I have sent a [proposal here](https://github.com/Expensify/App/issues/32658#issuecomment-1869488988) with this approach below. Please do let me know your thoughts on the proposal.

**Comment by mountiny** [comment]
I think the solution should be simpler and that would be to update the existing logic to fallback to the `lastMessageText` of the expense / iou report when the current logic evaluates to No activity

**Comment by rojiphil** [comment]
Thanks @mountiny for the response.

I agree that fallback to `lastMessageText` of the report will resolve the issue in LHN here. @paultsimura already mentioned this approach. So, I will pursue this only if I can better his proposal. 

Meanwhile, we do have scenarios where the FE display is affected due to insufficient data from BE. Here is one example as shown in the attached video where the participant details are not displayed in Report Preview.
 
Would a common solution like my [proposal here](https://github.com/Expensify/App/issues/32658#issuecomment-1869488988) help here? 
Curious to know your thoughts here.

https://github.com/Expensify/App/assets/3004659/c828e4bd-f528-4c14-9e32-8e806a060afd

**Comment by paultsimura** [comment]
@mountiny the `lastMessageText` is not localizable. That's the main concern that makes us look for (arguably) better solutions.
What's your take on this?

**Comment by mountiny** [comment]
We do already have performance problems tied to the amount of data in the app during app start or in general so I think thats our main reasoning to find other ways than providing more data in this case.

I would need to check into this more there has been couple similar issues at the moment admittedly and I am not sure if I understand the rootcause correctly here.

Could you summarize the root cause of this issue in the app (so what we are doing but is wrong because what data we are missing)

**Comment by rojiphil** [comment]
> We do already have performance problems tied to the amount of data in the app during app start or in general so I think thats our main reasoning to find other ways than providing more data in this case.
> Could you summarize the root cause of this issue in the app (so what we are doing but is wrong because what data we are missing)

@mountiny 
Let me try my best to explain.

As I understand, our earlier approach was to send most of the content in `openApp` response.  This would work just fine on app launch as almost all required data i.e. `Reports`, `Report Actions`, `Participants`, etc for the User are already present. However, performance problems arise as mentioned and I too agree that sending reduced data from BE will help in addressing this. 

Now, coming specifically to this issue, LHN message for a `Report Preview` depends on the child `IOU Report` as seen [here](https://github.com/Expensify/App/blob/7d1b0f8b99616c48aedb975dd73fc3086c9f20e3/src/libs/OptionsListUtils.js#L401). But, with the new approach of sending reduced data from BE, the child `IOU Report` is not present at all which is the problem here.  

From the solution perspective, we have already discussed some approaches. 

One other way to resolve this is to ensure that BE sends child `IOU Reports` too when the last visible action of a `Chat Report` is a `Report Preview`. But, I am not sure if BE should get into the nitty gritty of what is expected of the FE. 

The other alternative is my [proposal here](https://github.com/Expensify/App/issues/32658#issuecomment-1869488988). Since the User interacts primarily through the `Reports` displayed in LHN, I think it can be optimum to fetch all data for only these reports along with their child reports. This can resolve this issue and other similar issues too arising out of reduced data.

**Comment by mountiny** [comment]
Ok my take is that instead of some complicated solution we should just use the message from BE even if its not translated, we are talking about super edge case and english is the most common language so there is no problem with seeing it in english until you click into it. Additionally when you are in focus mode you can barely read it.

So I would say, lets just not translate it, show whatever is passed in the lastMessageText prop.

We will eventually have to handle server side internationalization and we can handle this then imho.

@rojiphil @paultsimura @cubuspl42 does that work?

**Comment by paultsimura** [comment]
Definitely works for me as it was a part of my proposal🙂

**Comment by rojiphil** [comment]
@mountiny 

While I agree that using the message from BE can solve the LHN issue here, there is another related issue of missing participant details as shown in the attachment below. This is also due to reduced data on app launch.
I mentioned this in my earlier [comment here](https://github.com/Expensify/App/issues/32658#issuecomment-1869698907)

The root cause for the related issue is the same as that of the LHN issue here. i.e. not fetching IOU Report in advance.  
How do we go about providing the missing participant data here without fetching the IOU Report in advance? 

![Screenshot 2023-12-27 at 10 29 37 PM](https://github.com/Expensify/App/assets/3004659/75c43209-012f-4f07-be7c-6b8bf5213f01)

**Comment by cubuspl42** [comment]
> So I would say, lets just not translate it, show whatever is passed in the lastMessageText prop.

I always tried to ensure things were localized, as I believed that's important for Expensify. There's even a checklist item for that. 🙃

But If you're confident that in this specific case, it's not worth the implementation effort, sure, we can skip it.

**Comment by rojiphil** [comment]
@mountiny 

Thanks for your patience and the response to my reasoning. 

As I understand, the approach we are taking is to find out the missing data for each issue and fix it. But, the approach I suggested is to optimally fetch the contents of reports early enough based on what FE needs. And, I don't understand correctly the advantage of our approach over my suggested approach.  

But, here are a few advantages I see in my suggested approach. 

1) Since we are fundamentally tackling the performance problem, here, the BE response to `openApp` can be super light by just sending only data of reports. That way, BE can send even more reduced data than what it is sending currently (e.g. do not send Report Actions at all in `openApp` response). Based on the FE analysis of the data of reports sent via `openApp`, FE can decide on the reports it is interested in and fetch them early enough. If required, sometime later on, we can also introduce `lastModifiedTime` in Reports and make it even more optimal.  

2) With the approach I suggested, we can also fix all such issues (like this one here) in one go. Otherwise, we will end up spending time on each such identified issue. And, I am not quite sure if we have to go this way. On the witty side though, this may be good news for all C and C+ contributors here including me. 

I would need your help understanding why we prefer the approach of finding missing data of individual issues and fixing it over my suggested approach.

**Comment by mountiny** [comment]
> Since we are fundamentally tackling the performance problem, here, the BE response to openApp can be super light by just sending only data of reports. That way, BE can send even more reduced data than what it is sending currently (e.g. do not send Report Actions at all in openApp response). Based on the FE analysis of the data of reports sent via openApp, FE can decide on the reports it is interested in and fetch them early enough. If required, sometime later on, we can also introduce lastModifiedTime in Reports and make it even more optimal.

that is being tackled by Report pagination which is a separate project going on in parallel

> I would need your help understanding why we prefer the approach of finding missing data of individual issues and fixing it over my suggested approach.

Maybe there is a misunderstanding, I think we all agree that we should just provide all the data needed from the BE and the FE should be just used to display that, ideally doing as little modification/ computation as possible.

However, that is not in scope of this issue and I think it should be handled holistically as part of the report pagination which will free up some of the performance bottleneck we have now and we can better load only what we need

For the purpose of this issue we can just fix this quickly so end user experience is not affected

**Comment by paultsimura** [comment]
Hi, since we've decided to go with falling back to the `lastMessageText`, could we re-review the proposals, please?

**Comment by cubuspl42** [comment]
I'm catching up...

We have two proposals:

- [Proposal](https://github.com/Expensify/App/issues/32658#issuecomment-1866134497) by @paultsimura 
- [Proposal](https://github.com/Expensify/App/issues/32658#issuecomment-1869488988) by @rojiphil

We decided to give up the localization in this case, and we could just accept the backend-provided English text.

We have feedback from Vit to reduce the scope of this issue, handling just its essence, and leaving the bigger picture changes to the report pagination project.

It feels like the proposal by @paultsimura is closer to our requirements, if I get things right.

@paultsimura 

Would you update your proposal to be in sync with the conclusions of the discussion? As is, it reasons about multiple options and raises localization concerns. Let's make it precise and up-to-date.

**Comment by paultsimura** [comment]
> Would you update your proposal to be in sync with the conclusions of the discussion?
Thanks, updated my proposal: https://github.com/Expensify/App/issues/32658#issuecomment-1866134497

**Comment by cubuspl42** [comment]
@paultsimura 

> ```diff
> -    } else {
> -        lastMessageTextFromReport = report ? report.lastMessageText || '' : '';
>     }
> -    return lastMessageTextFromReport;
> +    return lastMessageTextFromReport || lodashGet(report, 'lastMessageText', '');
> ```

How does the new code differ from the old one?

> After login, the report previews. are displayed incorrectly.

Incorrectly how? "Incorrectly" is the broadest of terms.

This section gives us the opportunity to add the technical perspective to the user-like perspective of the reporter from the QA team. It's often undervalued. It doesn't have to be long, a brief but accurate summary is what we aim for.

> It was decided to move away from loading all the reports data from BE, so this issue looks like a new feature request now.

This is not a root cause analysis at all. I understand that you mean that this is not a regression, but this section should be utilized to briefly explain why the app's code is not behaving as we want it to.

You're an experienced Contributor, and I know you often write proposals that could be used as an example for others. Let's keep it this way.

**Comment by paultsimura** [comment]
Thank you @cubuspl42, that's fair reasoning.
Updated the proposal: https://github.com/Expensify/App/issues/32658#issuecomment-1866134497
Also, I've put up a testing branch if you'd like to take a closer look: https://github.com/paultsimura/Expensify/tree/fix/32658-lazy-lhn-subtext

**Comment by cubuspl42** [comment]
Thank you.

Which logic ensures that `{user} owes {amount}` substitutes the display name (as expected) for `{user}` when it _does_ work, before your changes?

![image](https://github.com/Expensify/App/assets/2590174/a09eb21b-a6d7-4cda-a618-5abe01d962d3)

I understand that this is the expected for of this text ☝️

**Comment by paultsimura** [comment]
If the IOU report is loaded (i.e. when in was opened after relogin), my changes do not affect the flow – we build the message in the original way:

https://github.com/Expensify/App/blob/27a51a44e744f66956687550768e0037e6612923/src/libs/OptionsListUtils.js#L396-L405

We build the payer name for it here:

https://github.com/Expensify/App/blob/27a51a44e744f66956687550768e0037e6612923/src/libs/ReportUtils.ts#L2021-L2022

And the `getDisplayNameForParticipant` essentially uses `PersonalDetailsUtils.getDisplayNameOrDefault` which I suggest using for the replacement in the BE string as well:

https://github.com/Expensify/App/blob/27a51a44e744f66956687550768e0037e6612923/src/libs/ReportUtils.ts#L1455


And even though in the normal flow, the payer name is constructed of either `policyName` or `user.displayName`, we can safely use only users-based replacement, since the `policyName` is correctly returned in the message from the BE.

**Comment by cubuspl42** [comment]
I approve the proposal by @paultsimura

C+ reviewed 🎀 👀 🎀

**Comment by yuwenmemon** [comment]
We just got an assignee

**Comment by trjExpensify** [comment]
PR is in draft melv, what's the ETA on getting that into review @paultsimura? Thanks!

**Comment by paultsimura** [comment]
There is an ongoing discussion in Slack: https://expensify.slack.com/archives/C01GTK53T8Q/p1704394327411619
@mountiny @yuwenmemon could you please share what should be our behavior here based on the thread? 🙏

**Comment by paultsimura** [comment]
Melvin, we've just agreed on the expected BE behavior in Slack. I'm currently working on the PR, will be ready within an hour.

**Comment by trjExpensify** [comment]
👋 checklist time ahead of tomorrow!

**Comment by cubuspl42** [comment]
- The PR that introduced the bug has been identified. Link to the PR:
    -  https://github.com/Expensify/App/pull/10051
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
    - https://github.com/Expensify/App/pull/10051/files#r1466239591
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
    - No need for additional discussion
- Determine if we should create a regression test for this bug.
    - Up to the QA team
- If we decide to create a regression test for the bug, please propose the regression test steps to ensure the same bug will not reach production again.
  - Ensure that you have a 1:1 chat report where an IOU Preview is the last action
  - Log out
  - Log in
  - Verify that the summary text in the LHN for the mentioned report is "{userDisplayName} owes {amount}"

**Comment by cubuspl42** [comment]
@bernhardoj I believe that the "QA Steps" in your PR weren't up-to-date when we merged. My oversight. What do you think of the steps I suggested above?

**Comment by paultsimura** [comment]
> "QA Steps" in your PR weren't up-to-date when we merged

@cubuspl42 I was working on the PR for this issue. Are we talking about this one, or there's another? 🤔

**Comment by cubuspl42** [comment]
Sorry, I had Bernard's PR open in another tab and I confused usernames

**Comment by paultsimura** [comment]
> Log out
Log in
Open a chat with an IOU Report (but don't open the IOU Report itself).
 Verify that the report shows a correct message of "{user} owes {amount}" format in LHN

These are the steps from my PR. It's pretty similar to the ones you've added IMO. Would you prefer me to change it anyway?

**Comment by cubuspl42** [comment]
I'm just confirming that the latest are correct. In the one you quoted, we don't focus on the fact that IOU has to be the last item in the chat log.

**Comment by paultsimura** [comment]
Oh, I missed that, thanks. Will update now, sure!

**Comment by trjExpensify** [comment]
Confirming payments as follows: 

- $500 to @paultsimura for the fix 
- $500 to @cubuspl42 for the review.

**Comment by trjExpensify** [comment]
Settled up!
